### PR TITLE
Reform get neighbors

### PR DESCRIPTION
**Pull request by samwaseda**
I did what we discussed at the last pyiron meeting + I moved `shells` from the original `get_neighbors` algorithm to a function in `Neighbors` because this one appears to me more like an auxiliary attribute that can be processed afterwards.

### CODE REVIEW

**Comment by samwaseda** [comment]
I failed to copy DocString from `_get_neighbors` - help me!

**Comment by samwaseda** [comment]
Closes #942

**Review by liamhuber — COMMENTED** [review]
cf. docstring automation note, otherwise I'm happy 👍

**Comment by samwaseda** [comment]
Following @sudarsan-surendralal 's suggestion I'll separate `Neighbors` to a new file, so I make it a draft for now

File: pyiron/atomistics/structure/atoms.py [hunk 1]
```diff
@@ -1642,6 +1666,36 @@ def get_neighbors(
...
+            include_boundary=include_boundary,
+            exclude_self=exclude_self,
+            tolerance=tolerance,
+            id_list=id_list,
+        )
+        neigh.distances = np.array(neigh.distances)
+        neigh.vecs = np.array(neigh.vecs)
+        neigh.indices = np.array(neigh.indices)
+        return neigh
+
```
[the remark this answers is not in the document]
Comment (samwaseda) [reply pyiron/atomistics/structure/atoms.py:1718]: For some reason it doesn't work - I simply copied the content

[the remark this answers is not in the document]
Comment (liamhuber) [reply pyiron/atomistics/structure/atoms.py:1718]: Bizarre 😞

**Review by liamhuber — APPROVED** [review]
My requested change is apparently not working and anyhow not a big deal, so other than broken tests I'm content.

**Comment by samwaseda** [comment]
Actually it's still not fully done, but following a recent discussion I don't think a single pull request should become too long, so I stop at this point and open other pull requests.

**Comment by samwaseda** [comment]
I do not really understand why this one test is failing, but if there’s no objection, I would love to merge this one by tonight.

**Pull request merged**